server/src/input/pdfminer/PdfminerExtractor.ts

```diff
@@ -27,17 +30,70 @@ import * as pdfminer from './pdfminer';
  * information in a clever way.
  */
 export class PdfminerExtractor extends Extractor {
-  public run(inputFile: string): Promise<Document> {
+  public async run(inputFile: string): Promise<Document> {
     return utils.repairPdf(inputFile).then((repairedPdf: string) => {
-      const pdfminerExtract: Promise<Document> = pdfminer.execute(repairedPdf);
-
-      const extractFont = extractImagesAndFonts(repairedPdf);
-      return Promise.all([pdfminerExtract, extractFont]).then(
-        ([doc, assetsFolder]: [Document, string]) => {
-          doc.assetsFolder = assetsFolder;
-          return doc;
-        },
-      );
+      return this.pageNumber(repairedPdf).then(totalPages => {
+        logger.info(
+          'Extracting contents (' +
+            totalPages.toString() +
+            " pages) with pdfminer's pdf2txt.py tool...",
+        );
+        const startTime: number = Date.now();
+        const extractFont = extractImagesAndFonts(repairedPdf);
+        const pdfminerExtract = this.extractFile(repairedPdf, 1, 500, totalPages);
```

Comment:
Maybe place the value of `500` as a clearly visible `constant` at top of file or even a config parameter with a sensible default value ?